R/setLabels.R

```diff
@@ -0,0 +1,104 @@
+#' Ensure that hyperSpec and non hyperSpec objects have the correct labels
+
+#' @param .data hyperSpec object
+#' @param ... resulting data frame
+#' @return Object with the correct labels
+#' @md
+#'
+#' @importFrom hyperSpec chk.hy
+#' @importFrom hyperSpec labels labels<-
+#' @importFrom rlang enquos
+#' @importFrom rlang quo_name
+#' @importFrom rlang quo_get_expr
+#'
+#' @examples
+#' flu %>%
+#'  setLabels(.wavelength = "f / THz", c = "c / ml")
+#'
+#' @export
+setLabels <- function(.data, ...) {
+
+    # Check if user passed in a hyperSpec object
+    chk.hy(.data)
+    args <- enquos(...)
+    args_names <- names(args)
+    labels2update <- args_names[args_names %in% names(labels(.data))]
+    if (length(args) == 0L) {
+      return(NULL)
+    }
+    for (i in seq_along(args)) {
+        setlabels2 <- quo_name(quo_get_expr(args[[i]]))
+        if (grepl('expr', setlabels2)) {
+            setlabels2 <- as.expression(setlabels2)
+        }
+        labels(.data, labels2update[i]) <- setlabels2
+    }
+    .data
+}
+
+#' @rdname setLabels
+#' @aliases setLabels
+#' @param .data hyperSpec object
+#' @param ... list of columns to update on
+#' @export
+setLabels.transmute <- function(.data, ...) {
+    setLabels(.data, ...)
+}
+
+#' @rdname setLabels
+#' @aliases setLabels
+#' @param .data hyperSpec object
+#' @param res resulting data frame
+#' @export
+setLabels.select <- function(.data, res) {
```

Comment:
- Should `setLabels.select()` and `setLabels.rename()` really be exported?
- since `method.class()` signifies S3 methods, would snake-case names (`setLabels_select()`) be better here?

Reply:
Okay, will change to snake-case